test/addons-napi/test_general/testFinalizer.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+// Flags: --expose-gc
+
+const common = require('../../common');
+const test_general = require(`./build/${common.buildType}/test_general`);
+const assert = require('assert');
+
+let finalized = {};
+const callback = common.mustCall(2);
+
+// Add two items to be finalized and ensure the callback is called for each.
+test_general.addFinalizerOnly(finalized, callback);
+test_general.addFinalizerOnly(finalized, callback);
+
+// Ensure attached items cannot be retrieved.
+common.expectsError(() => test_general.unwrap(finalized),
+                    { type: Error, message: 'Invalid argument' });
+
+// Ensure attached items cannot be removed.
+common.expectsError(() => test_general.removeWrap(finalized),
+                    { type: Error, message: 'Invalid argument' });
+finalized = null;
+global.gc();
```

Comment:
Do we also need to test that you can add a finalizer to an object that is already wrapped?

Reply:
I have now added such a test.